### ISSUE TITLE
Can't custom JsonConverter with constructor arguments. Error "Generators with constructor arguments are not supported"

### ISSUE THREAD

**Issue by phuongvt** [google/json_serializable.dart#1373]
When I write DateTimeConverter extends from JsonConverter with constructor arguments. Then use

```
  @JsonKey(name: 'time')
  @DateTimeConverter(pattern: 'YYYY/MM/dd')
  final DateTime time;
```

`run build_runner build` print error "Generators with constructor arguments are not supported"

**Comment by faisalansari0367** [comment]
@phuongvt Thanks for this awesome PR.
When can we expect this into release?

**Comment by phuongvt** [comment]
@faisalansari0367 : Thank you for your support!

First of all, sorry for the late reply.
In my case, I need to define many parameters in JsonConverter structure, and the parameters have too many different values because they are of type String => there can be many "named constructors" that need to be created without rarely reused. For example:

```
pattern= 'YYYY/MM/dd'
pattern= 'YYYY/MM/dd HH:mm:ss'
pattern= 'YYYY/MM/dd HH:mm'
pattern= 'YYYY-MM-dd'
...
```

so I still find "Generators with constructor arguments" necessary. I need MR as soon as possible to support my projects

Thank you!

**Comment by phuongvt** [comment]
@kevmoo yes, I was considering creating 'named constructors' that are often reused. However, I still think supporting 'Generators with constructor arguments' is necessary

**Comment by kevmoo** [comment]
> @kevmoo yes, I was considering creating 'named constructors' that are often reused. However, I still think supporting 'Generators with constructor arguments' is necessary

This will add non-trivial complexity to this package, sadly.

**Comment by phuongvt** [comment]
@kevmoo What do you think if I create a new class ArgumentsJsonConverter. With JsonConverter, I will throw an exception "Use ArgumentsJsonConverter to support constructor arguments or define named constructors ([#1373](https://github.com/google/json_serializable.dart/issues/1373))" instead of "Generators with constructor arguments are not supported"

**Comment by MoralCode** [comment]
i believe I have found a convenient enough workaround.

in your Converter:
```dart
  const DecimalIntConverter.tenths() : this(places: -1);
  const DecimalIntConverter({this.places = 0, this.base = 10});
```

in your type:
```dart
   @DecimalIntConverter.tenths() double someValue;
```

**Comment by amrgetment** [comment]
This works fine for me
https://github.com/google/json_serializable.dart/issues/1373#issuecomment-1827998656
I think macros could solve this case too 🤔
